pkg/loadbalance/weighted_round_robin.go

```diff
@@ -46,16 +46,18 @@ func newWeightedRoundRobinPicker(instances []discovery.Instance) Picker {
 	wrrp.nodes = make([]*wrrNode, wrrp.size)
 	offset := fastrand.Uint64n(wrrp.size)
 	totalWeight := 0
+	gcd := 0
 	for idx := uint64(0); idx < wrrp.size; idx++ {
 		ins := instances[(idx+offset)%wrrp.size]
 		totalWeight += ins.Weight()
+		gcd = gcdInt(gcd, ins.Weight())
 		wrrp.nodes[idx] = &wrrNode{
 			Instance: ins,
 			current:  0,
 		}
 	}
 
-	wrrp.vcapacity = uint64(totalWeight)
+	wrrp.vcapacity = uint64(totalWeight / gcd)
```

Comment:
it's not very safe to assume wrrp.size will not be `0`. may be need to check if gcd is zero

Reply:
Hello, @joway. If `wrrp.size` could be `0`, maybe the shuffle will panic too. And there is no other caller of private function `newWeightedRoundRobinPicker` but `createPicker`. So I think it's safe.
https://github.com/cloudwego/kitex/blob/b110a7fc64eacdc85e3224ca1eb27cb8dddddc2b/pkg/loadbalance/weighted_round_robin.go#L47

Reply:
Can you write a test case for this logic? For example, test `newWeightedRoundRobinPicker` func and check the return *WeightedRoundRobinPicker's vcapacity

Reply:
Yes, I will do this later.

Reply:
> Can you write a test case for this logic?

Done